Code/Mantid/MantidPlot/src/CurvesDialog.cpp

```diff
@@ -451,19 +451,24 @@ bool CurvesDialog::addCurve(const QString& name)
   return false;
 }
 
+
+/**Remove curves function
+*
+*/
 void CurvesDialog::removeCurves()
 {
   int count = contents->count();
+  QList<QListWidgetItem *> lst = contents->selectedItems();
+  
   //disables user from deleting last graph from the graph
-  if(count == 1) {
+  if(count == 1 || count == lst.size()) {
   QMessageBox::information( 
```

Comment:
You could use [`QMessageBox::warning`](http://doc.qt.io/qt-4.8/qmessagebox.html#warning) here instead, then the title would  only need to read "Cannot Delete".